modules/carto/src/layers/carto-binary-tile-layer.js

```diff
@@ -0,0 +1,56 @@
+/* global TextDecoder */
+import {log} from '@deck.gl/core';
+import {ClipExtension} from '@deck.gl/extensions';
+import {MVTLayer} from '@deck.gl/geo-layers';
+import {GeoJsonLayer} from '@deck.gl/layers';
+import {geojsonToBinary} from '@loaders.gl/gis';
+
+function parseJSON(arrayBuffer) {
+  return JSON.parse(new TextDecoder().decode(arrayBuffer));
+}
+
+function parseCartoBinaryTile(arrayBuffer, options) {
+  if (!arrayBuffer) return null;
+  const formatTiles = (options && options.formatTiles) || 'geojson';
+  if (formatTiles === 'geojson') return geojsonToBinary(parseJSON(arrayBuffer).features);
+  log.assert(formatTiles === 'geojson', `formatTiles must be geojson`);
+  return null;
+}
+
+const CartoBinaryTileLoader = {
+  name: 'CARTO Binary Tile',
+  id: 'cartoBinaryTile',
+  module: 'carto',
+  extensions: ['pbf'],
+  mimeTypes: ['application/x-protobuf'],
+  category: 'geometry',
+  worker: false,
+  parse: async (arrayBuffer, options) => parseCartoBinaryTile(arrayBuffer, options),
+  parseSync: parseCartoBinaryTile
+};
+
+// Currently we only support loading via geojson, but in future the data
+// format will be binary, as such keep `binary` in layer name
+export default class CartoBinaryTileLayer extends MVTLayer {
```

Comment:
Can we rename this to `CartoDynamicTileLayer`?

Reply:
I think it fits better the purpose, the format of the tiles could be a prop of this layer

Reply:
I did think that, but have we also not considered that in the future we might use the same data format for tile layers? As such, the tile might not always be dynamic. At any rate, the client doesn't really care if the backend is generating the tiles dynamically or not. So perhaps `CartoTileLayer` is better? 

I will add the format of the tiles as a prop.

Reply:
Sounds good.

Reply:
After a second thought, I've changed my mind :). CartoTileLayer could be confused with CartoLayer + tileset. As this layer is not exposed externally, it's only used by us internally, I think CartoDynamicTyleLayer makes sense because it has the format and limitations (as the clipping) of our dynamic tiling backend.